airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/legacy.py

```diff
@@ -0,0 +1,146 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from functools import lru_cache
+from typing import Any, Iterable, List, Mapping, Optional, Tuple, Union
+
+from airbyte_cdk.models import AirbyteStream, SyncMode
+from airbyte_cdk.sources import Source
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.availability_strategy import AvailabilityStrategy
+from airbyte_cdk.sources.streams.concurrent.abstract_stream import AbstractStream
+from airbyte_cdk.sources.streams.concurrent.availability_strategy import AbstractAvailabilityStrategy
+from airbyte_cdk.sources.streams.concurrent.exceptions import ExceptionWithDisplayMessage
+from airbyte_cdk.sources.streams.core import StreamData
+from airbyte_cdk.sources.utils.slice_logger import SliceLogger
+from deprecated.classic import deprecated
+
+"""
+This module contains adapters to help enabling concurrency on existing Stream without needing to migrate to AbstractStream
+"""
+
+
+@deprecated("This class is experimental. Use at your own risk.")
+class StreamFacade(Stream):
+    """
+    The StreamFacade is a Stream that wraps an AbstractStream and exposes it as a Stream.
+
+    All methods either delegate to the wrapped AbstractStream or provide a default implementation.
+    The default implementations define restrictions imposed on Streams migrated to the new interface. For instance, only source-defined cursors are supported.
+    """
+
+    def __init__(self, stream: AbstractStream):
+        """
+        :param stream: The underlying AbstractStream
+        """
+        self._stream = stream
+
+    def read_full_refresh(
```

Comment:
Should this be private?

Reply:
I don't think this method can be private because [it's called from the abstract source](https://github.com/airbytehq/airbyte/pull/30111/files#diff-51c861828e0a614ef6ee21390f3ba17d7138549f0575051fdb24868ee05eabbbR292)